File: web/admin/src/store/entities.ts
export type ProviderRecord = {
  id: string;
  name: string;
  type: string;
  base_url?: string;
  baseURL: string;
  apiKey?: string;
  organization_id?: string;
  enabled: boolean;
  status: "healthy" | "warning" | "disabled" | "active" | string;
  priority: number;
  models_json?: string;
  models: string[];
  rate_limit_rpm?: number;
  rate_limit_tpm?: number;
  rpm: number;
  tpm: number;
};

export type LocalKeyRecord = {
  id: string;
  name: string;
  display_key?: string;
  displayKey: string;
  allowed_models_json?: string;
  allowed_providers_json?: string;
  allowedModels: string[];
  allowedProviders: string[];
  monthly_budget?: number;
  monthlyBudget: number;
  current_spend?: number;
  currentSpend: number;
  token_budget?: number;
  tokenBudget: number;
  current_tokens?: number;
  currentTokens: number;
  enabled: boolean;
  revoked_at?: string | null;
  expires_at?: string | null;
  status: "active" | "warning" | "revoked";
};

export type RoutingRuleRecord = {
  id: string;
  model_pattern?: string;
  modelPattern: string;
  strategy: string;
  provider_chain?: string;
  providerChain: string[];
  fallback_chain?: string;
  fallbackChain: string[];
  enabled: boolean;
};

export type ModelAliasRecord = {
  id?: string;
  alias: string;
  target: string;
  fallback_chain?: string;
  fallbackChain: string[];
};

export type DistributionPlanRecord = {
  package_name?: string;
  mode?: string;
  includes?: string[];
  notes?: string[];
};

export type RoutingSimulation = {
  model: string;
  key: string;
  format: string;
  target: string;
  fallback: string;
  cost: string;
  ttft: string;
};

export type SettingsRecord = {
  host: string;
  port: number;
  admin_path?: string;
  adminPath: string;
  admin_username?: string;
  adminUsername: string;
  theme: "light" | "dark" | "system" | string;
  update_channel?: string;
  updateChannel: string;
  backup_interval?: string;
  backupInterval: string;
  log_level?: string;
  logLevel: string;
  retention_days?: number;
  retentionDays: number;
  bundle_mode?: string;
  bundleMode: string;
};

export const providerRecords: ProviderRecord[] = [
  {
    id: "prov-openai-primary",
    name: "OpenAI Primary",
    type: "openai",
    baseURL: "https://api.openai.com",
    enabled: true,
    status: "healthy",
    priority: 1,
    models: ["gpt-4o", "gpt-4o-mini", "o3-mini"],
    rpm: 120,
    tpm: 400000
  },
  {
    id: "prov-claude-premium",
    name: "Claude Premium",
    type: "anthropic",
    baseURL: "https://api.anthropic.com",
    enabled: true,
    status: "healthy",
    priority: 2,
    models: ["claude-sonnet-4", "claude-haiku-4"],
    rpm: 90,
    tpm: 320000
  },
  {
    id: "prov-deepseek-saver",
    name: "DeepSeek Saver",
    type: "deepseek",
    baseURL: "https://api.deepseek.com",
    enabled: true,
    status: "warning",
    priority: 3,
    models: ["deepseek-chat", "deepseek-reasoner"],
    rpm: 160,
    tpm: 600000
  }
];

export const localKeyRecords: LocalKeyRecord[] = [
  {
    id: "key-codex",
    name: "Codex Pro",
    displayKey: "lg-a3f8****9k2m",
    allowedModels: ["gpt-4o", "claude-sonnet-4"],
    allowedProviders: ["OpenAI Primary", "Claude Premium"],
    monthlyBudget: 150,
    currentSpend: 42.8,
    tokenBudget: 10000000,
    currentTokens: 2340000,
    enabled: true,
    status: "active"
  },
  {
    id: "key-cursor",
    name: "Cursor Team",
    displayKey: "lg-b8d2****c1q9",
    allowedModels: ["gpt-4o-mini", "deepseek-chat"],
    allowedProviders: ["OpenAI Primary", "DeepSeek Saver"],
    monthlyBudget: 80,
    currentSpend: 28.1,
    tokenBudget: 8500000,
    currentTokens: 1720000,
    enabled: true,
    status: "warning"
  }
];

export const routingRuleRecords: RoutingRuleRecord[] = [
  {
    id: "route-gpt4o",
    modelPattern: "gpt-4o*",
    strategy: "Priority + Fallback",
    providerChain: ["OpenAI Primary", "Azure Backup"],
    fallbackChain: ["Azure Backup", "OpenRouter"],
    enabled: true
  },
  {
    id: "route-claude",
    modelPattern: "claude-*",
    strategy: "Priority + Fallback",
    providerChain: ["Claude Premium", "OpenRouter"],
    fallbackChain: ["OpenRouter", "DeepSeek Saver"],
    enabled: true
  }
];

export const settingsRecord: SettingsRecord = {
  host: "127.0.0.1",
  port: 18743,
  adminPath: "/admin",
  adminUsername: "admin",
  theme: "system",
  updateChannel: "stable",
  backupInterval: "24h",
  logLevel: "standard",
  retentionDays: 30,
  bundleMode: "single-binary"
};

File: internal/models/entities.go
package models

import "time"

type Provider struct {
	ID              string     `json:"id" gorm:"primaryKey"`
	Name            string     `json:"name"`
	Type            string     `json:"type"`
	BaseURL         string     `json:"base_url"`
	APIKeyEncrypted string     `json:"-"`
	OrganizationID  string     `json:"organization_id"`
	Enabled         bool       `json:"enabled"`
	Priority        int        `json:"priority"`
	Status          string     `json:"status"`
	ModelsJSON      string     `json:"models_json"`
	RateLimitRPM    int        `json:"rate_limit_rpm"`
	RateLimitTPM    int        `json:"rate_limit_tpm"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
	LastUsedAt      *time.Time `json:"last_used_at,omitempty"`
	DeletedAt       *time.Time `json:"deleted_at,omitempty"`
}

type LocalKey struct {
	ID                   string     `json:"id" gorm:"primaryKey"`
	Name                 string     `json:"name"`
	KeyHash              string     `json:"-"`
	DisplayKey           string     `json:"display_key" gorm:"-"`
	AllowedModelsJSON    string     `json:"allowed_models_json"`
	AllowedProvidersJSON string     `json:"allowed_providers_json"`
	MonthlyBudget        float64    `json:"monthly_budget"`
	CurrentSpend         float64    `json:"current_spend"`
	TokenBudget          int64      `json:"token_budget"`
	CurrentTokens        int64      `json:"current_tokens"`
	Enabled              bool       `json:"enabled"`
	ExpiresAt            *time.Time `json:"expires_at,omitempty"`
	CreatedAt            time.Time  `json:"created_at"`
	UpdatedAt            time.Time  `json:"updated_at"`
	LastUsedAt           *time.Time `json:"last_used_at,omitempty"`
	RevokedAt            *time.Time `json:"revoked_at,omitempty"`
}

type RoutingRule struct {
	ID            string    `json:"id" gorm:"primaryKey"`
	ModelPattern  string    `json:"model_pattern"`
	Strategy      string    `json:"strategy"`
	ProviderChain string    `json:"provider_chain"`
	FallbackChain string    `json:"fallback_chain"`
	Enabled       bool      `json:"enabled"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type ModelAlias struct {
	ID            string    `json:"id" gorm:"primaryKey"`
	Alias         string    `json:"alias"`
	Target        string    `json:"target"`
	FallbackChain string    `json:"fallback_chain"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type Setting struct {
	Key       string    `json:"key" gorm:"primaryKey"`
	ValueJSON string    `json:"value_json"`
	UpdatedAt time.Time `json:"updated_at"`
}

type UsageRecord struct {
	ID             string    `json:"id" gorm:"primaryKey"`
	LocalKeyID     string    `json:"local_key_id"`
	ProviderID     string    `json:"provider_id"`
	ModelRequested string    `json:"model_requested"`
	ModelActual    string    `json:"model_actual"`
	APIFormat      string    `json:"api_format"`
	InputTokens    int64     `json:"input_tokens"`
	OutputTokens   int64     `json:"output_tokens"`
	TotalCostUSD   float64   `json:"total_cost_usd"`
	LatencyMS      int64     `json:"latency_ms"`
	Success        bool      `json:"success"`
	CreatedAt      time.Time `json:"created_at"`
}

type RequestLog struct {
	ID           string    `json:"id" gorm:"primaryKey"`
	LocalKeyID   string    `json:"local_key_id"`
	ProviderID   string    `json:"provider_id"`
	Path         string    `json:"path"`
	Method       string    `json:"method"`
	StatusCode   int       `json:"status_code"`
	LatencyMS    int64     `json:"latency_ms"`
	ErrorMessage string    `json:"error_message"`
	MetadataJSON string    `json:"metadata_json"`
	CreatedAt    time.Time `json:"created_at"`
}
